desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -79,14 +80,64 @@ def parse_events(request, event):
   """
   channel_id = event.get('channel')
   user_id = event.get('user')
-  message_element = event['blocks'][0]['elements'] if event.get('blocks') else []
+  message_element = event['blocks'][0].get('elements', []) if event.get('blocks') else []
 
   if event.get('type') == 'message':
     handle_on_message(request.get_host(), request.is_secure(), channel_id, event.get('bot_id'), message_element, user_id)
 
   if event.get('type') == 'link_shared':
     handle_on_link_shared(request.get_host(), channel_id, event.get('message_ts'), event.get('links'), user_id)
 
+  if event.get('type') == 'app_mention':
+    handle_on_app_mention(request.get_host(), channel_id, user_id, event.get('text'))
+
+
+def handle_on_app_mention(host_domain, channel_id, user_id, text):
+  if text and 'query bank' in text:
+    slack_user = check_slack_user_permission(host_domain, user_id)
+    user = get_user(channel_id, slack_user)
+
+    handle_query_bank(channel_id, user_id)
+
+  
+def handle_query_bank(channel_id, user_id):
+  with open(os.path.join(os.path.dirname(__file__), 'data/queries.json')) as file:
+    data = json.load(file)
```

Comment:
All the query bank management should be out in his own utils not tangle with slack logic, e.g.

get_query_suggestions(dialect, query_search) --> [] of queries

(in the new `assistant` directory)